src/Pim/Component/Catalog/Updater/AttributeOptionUpdater.php

```
@@ -56,14 +56,9 @@ public function update($attributeOption, array $data, array $options = [])
             );
         }
 
-        $isNew = $attributeOption->getId() === null;
-        $readOnlyFields = ['attribute', 'code'];
         foreach ($data as $field => $value) {
             $this->validateDataType($field, $value);
-            $isReadOnlyField = in_array($field, $readOnlyFields);
-            if ($isNew || !$isReadOnlyField) {
-                $this->setData($attributeOption, $field, $value);
-            }
+            $this->setData($attributeOption, $field, $value);
```

Before modification: 
When you modify the optionCode or attributeCode of an existing attribute option, it was silently ignored (and then not modified).

After modification:
When you modify the optionCode or attributeCode of an existing attribute option, validators raised an exception (because it's immutable properties).

I did that to make consistent the behaviour between the entities. Properties should not be ignored in the updaters if they are immutable.